File: TimerCollapsibleList-Objective-C/TimerCollapsibleList-Objective-C/DataSource.m
//
//  DataSource.m
//  Objective-C-List
//
//  Created by Evan Wang on 2022/3/10.
//

#import "DataSource.h"
@implementation DataSource

- (instancetype)init {
    self = [super init];
    if (self) {
        NSDate *date = [NSDate now];
        NSTimeInterval hour = 60 * 60;
        NSTimeInterval day = 60 * 60 * 24;
        NSMutableArray<SectionItem*> *sectionArray = [NSMutableArray array];
        for (int i = 0; i < 10; i ++) {
            NSMutableArray<CellItem*> *itemArray =[NSMutableArray array];
            for (int j = 0; j < 10; j ++) {
                CellItem *item = [CellItem new];
                item.date = [date dateByAddingTimeInterval: day * i + hour * j];
                [itemArray addObject: item];
            }
            SectionItem *section = [SectionItem new];
            section.date = [date dateByAddingTimeInterval:day * i];
            section.items = [NSArray arrayWithArray:itemArray];
            [sectionArray addObject:section];
        }
        _sections = [NSArray arrayWithArray:sectionArray];
    }
    return self;
}
@end

File: TimerCollapsibleList-Swift/Swift/DataSource.swift
//
//  DataSource.swift
//  TimerCollapsibleList-Swift
//
//  Created by Evan Wang on 2022/3/10.
//

import Foundation

class CellItem {
    let date: Date
    
    init(date: Date) {
        self.date = date
    }
}

class SectionItem {
    let items: [CellItem]
    let date: Date
    init(date: Date, items: [CellItem]) {
        self.items = items
        self.date = date
    }
}

class DataSource {
    let sections: [SectionItem]
    
    init() {
        let date = Date()
        let hour:TimeInterval = 60 * 60
        let day:TimeInterval = 60 * 60 * 24
        var _sections:[SectionItem] = []
        for i in 1...10 {
            var items: [CellItem] = []
            for j in 1...10 {
                let item = CellItem(date: date.addingTimeInterval( Double(i) * day + Double(j) * hour))
                items.append(item)
            }
            let section = SectionItem(date: date.addingTimeInterval(Double(i) * day),items: items)
            _sections.append(section)
        }
        self.sections = _sections
    }
}
